lib/timezone/timezone.ex

```diff
@@ -77,6 +77,19 @@ defmodule Timex.Timezone do
   def name_of("Y"),    do: name_of(-12)
   def name_of("Z"),    do: "UTC"
   def name_of("UT"),   do: "UTC"
+  def name_of("+0330"), do: "Asia/Tehran"
```

Comment:
Yeah I totally understand :), you've been on fire the last few days anyway haha.

This particular issue is something I've thought about a few times, and I have mixed feelings. On the one hand, I want to avoid supporting edge cases not handled by the Olson database, on the other, I want to be pragmatic about supporting real world scenarios. I was mostly waiting for exactly this situation, someone encounters it and requests a fix. You took it a step further and actually implemented a fix, which is awesome! Since we're here though, might as well take it all the way :)